core/model/src/main/java/org/eclipse/rdf4j/model/util/Models.java

```diff
@@ -753,4 +758,152 @@ private static Model toModel(Iterable<? extends Statement> iterable) {
 	public static Model synchronizedModel(Model toSynchronize) {
 		return new SynchronizedModel(toSynchronize);
 	}
+
+	/**
+	 * Converts the supplied RDF* model to RDF reification statements. The converted statements are sent to the supplied
+	 * consumer function.
+	 * <p>
+	 * The supplied value factory is used to create all new statements.
+	 *
+	 * @param vf       the {@link ValueFactory} to use for creating statements.
+	 * @param model    the {@link Model} to convert.
+	 * @param consumer the {@link Consumer} function for the produced statements.
+	 */
+	public static void convertRDFStarToReification(ValueFactory vf, Model model, Consumer<Statement> consumer) {
+		model.forEach(st -> Statements.convertRDFStarToReification(vf, st, consumer));
+	}
+
+	/**
+	 * Converts the supplied RDF* model to RDF reification statements. The converted statements are sent to the supplied
+	 * consumer function.
+	 *
+	 * @param model    the {@link Model} to convert.
+	 * @param consumer the {@link Consumer} function for the produced statements.
+	 */
+	public static void convertRDFStarToReification(Model model, Consumer<Statement> consumer) {
+		convertRDFStarToReification(SimpleValueFactory.getInstance(), model, consumer);
+	}
+
+	/**
+	 * Converts the supplied RDF* model to an RDF reification model.
+	 * <p>
+	 * The supplied value factory is used to create all new statements.
+	 *
+	 * @param vf    the {@link ValueFactory} to use for creating statements.
+	 * @param model the {@link Model} to convert.
+	 * @return the converted {@link Model}.
+	 */
+	public static Model convertRDFStarToReification(ValueFactory vf, Model model) {
+		Model reificationModel = new LinkedHashModel();
+		convertRDFStarToReification(vf, model, reificationModel::add);
+		return reificationModel;
+	}
+
+	/**
+	 * Converts the supplied RDF* model to an RDF reification model.
+	 *
+	 * @param model the {@link Model} to convert.
+	 * @return the converted {@link Model}.
+	 */
+	public static Model convertRDFStarToReification(Model model) {
+		return convertRDFStarToReification(SimpleValueFactory.getInstance(), model);
+	}
+
+	/**
+	 * Converts the supplied RDF reification model to RDF* statements. The converted statements are sent to the supplied
+	 * consumer function.
+	 * <p>
+	 * The supplied value factory is used to create all new statements.
+	 *
+	 * @param vf       the {@link ValueFactory} to use for creating statements.
+	 * @param model    the {@link Model} to convert.
+	 * @param consumer the {@link Consumer} function for the produced statements.
+	 */
+	public static void convertReificationToRDFStar(ValueFactory vf, Model model, Consumer<Statement> consumer) {
+		Map<Resource, Triple> convertedStatements = new HashMap<>();
+		model.filter(null, RDF.TYPE, RDF.STATEMENT).forEach((s) -> {
+			Value subject = object(model.filter(s.getSubject(), RDF.SUBJECT, null)).orElse(null);
```

Comment:
This kind of use of .filter(....) shows that we really need a .select(...) instead. I think .filter(....) is too heavyweight with all its guarantees about returning a model that reflects the underlying model...while all we really need here is at iterator.

Reply:
I think it might be faster to have a java switch over `model.filter(s.getSubject(), null, null)` instead of doing a model.filter for each.

Reply:
> This kind of use of .filter(....) shows that we really need a .select(...) instead. I think .filter(....) is too heavyweight with all its guarantees about returning a model that reflects the underlying model...while all we really need here is at iterator.

Agreed. Issue logged: https://github.com/eclipse/rdf4j/issues/2030